api/models/groups.py

```diff
@@ -0,0 +1,82 @@
+from django.db import models
+
+from api.models.base import GroupOrPerson
+from api.models.constants import DEFAULT_CHOICE_CHAR_FIELD_MAX_LENGTH
+from api.models.days_time_slots import DayAndTimeSlot
+from api.models.languages_levels import Language, LanguageAndLevel
+from api.models.people import Coordinator, Student, Teacher, TeacherUnder18
+
+
+class GroupCommon(GroupOrPerson):
+    """Abstract model for attributes shared by regular groups and speaking clubs."""
+
+    coordinators = models.ManyToManyField(Coordinator)
+    students = models.ManyToManyField(Student)
+    teachers = models.ManyToManyField(Teacher)
+    # group chat created manually by the coordinator/teacher
+    telegram_chat_url = models.URLField(null=True, blank=True)
+
+    class Meta:
+        abstract = True
+
+
+# TODO think about group clusters for auto-constructed groups
+class Group(GroupCommon):
+    """Model for a regular language group."""
+
+    class Status(models.TextChoices):
+        PENDING = "pending", "Pending"
+        STUDYING = "study", "Studying"
+        FINISHED = "finish", "Finished"
+        # TODO put statuses here once they are finalized
+
+    availability_slot = models.ManyToManyField(DayAndTimeSlot)
+    is_for_staff_only = models.BooleanField(default=False)
+    language_and_level = models.ForeignKey(LanguageAndLevel, on_delete=models.PROTECT)
+    lesson_duration = models.PositiveSmallIntegerField()
+    status = models.CharField(
+        max_length=DEFAULT_CHOICE_CHAR_FIELD_MAX_LENGTH, choices=Status.choices
+    )
+    start_date = models.DateField(null=True, blank=True)
+    # this field could be useful for overview, but can be filled automatically when
+    # a corresponding log event is created:
+    end_date = models.DateField(null=True, blank=True)
+
+    # some research showed that it's better to store the schedule not in a single text field
+    # with some pre-defined syntax, but in 7 columns, one per day of the week
+    monday = models.TimeField(null=True, blank=True)
+    tuesday = models.TimeField(null=True, blank=True)
+    wednesday = models.TimeField(null=True, blank=True)
+    thursday = models.TimeField(null=True, blank=True)
+    friday = models.TimeField(null=True, blank=True)
+    saturday = models.TimeField(null=True, blank=True)
+    sunday = models.TimeField(null=True, blank=True)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=["telegram_chat_url"], name="telegram_chat_url")
+        ]
+
```

Comment:
maybe another constraint should be there stating that at least one of day-of-the-week columns must have something in it, but it's kind of complicated for me right now :)

Reply:
You could try to use "CheckConstratint" and check it there with kind of `CheckConstraint(Q(monday__isnull=False) | Q(tuesday__isnull=False) ...)`

Reply:
Done